core/src/main/scala/org/apache/spark/storage/StorageLevel.scala

```diff
@@ -86,7 +83,7 @@ class StorageLevel private(
       false
   }
 
-  def isValid: Boolean = (useMemory || useDisk || useOffHeap) && (replication > 0)
+  def isValid: Boolean = (useMemory || useDisk) && (replication > 0)
```

Comment:
Do we also want to check that if `offHeap` is true then `useMemory` and `useDisk` must also be true?